docs/development/doc_guide.rst

```diff
@@ -1,76 +1,849 @@
+is made possible with the `sphinx.ext.intersphinx` extension.
+
+When we include ``astropy.units.Quantity`` in the documentation, it will
+show up as `astropy.units.Quantity` with a link to the appropriate page
+in Astropy documentation. Similarly, ``~astropy.units.Quantity`` will
+show up as `~astropy.units.Quantity`.
+
+The mappings are defined in the :confval:`intersphinx_mapping`
+dictionary in `docs/conf.py`_, and include Python_, NumPy_, SciPy_,
+Astropy_, pandas_, Sphinx_, and `sphinx_automodapi`_.
+
+New source packages may be added, but please verify that references to a
+function or class in that package show up correctly in `PlasmaPy's
+documentation`_. The name of the package does not always link as
+expected.
+
+.. hint::
+
+   If a cross-link is not working as expected this is usually due to one
+   of the following reasons:
+
+   * A typo;
+   * The package not being defined in :confval:`intersphinx_mapping`, or
+   * The referenced source package not properly or fully indexing their
+     own code, which is common in Python_ packages.
+
+Substitutions
+~~~~~~~~~~~~~
+
+Some functions and classes are referred to repeatedly throughout the
+documentation. reST_ allows us to `define substitutions
+<https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#substitution-definitions>`_.
+
+.. code-block:: rst
+
+   .. |Particle| replace:: `~plasmapy.particles.particle_class.Particle`
+
+PlasmaPy has certain common substitutions pre-defined so that they can
+be used elsewhere in the documentation. For example, we can write
+``|Quantity|`` instead of ``~astropy.units.Quantity``, and
+``|Particle|`` instead of
+``~plasmapy.particles.particle_class.Particle``.  For an up-to-date list
+of substitutions, please refer to the `docs/common_links.rst`_ file.
+
+Since substitutions are performed by Sphinx_ when the documentation is
+built, any substitution used in docstrings will not show up when using
+Python's `help` function (or the like). For example, when ``|Particle|``
+is used in a docstring, `help` will show it as ``|Particle|`` rather
+than ``~plasmapy.particles.particle_class.Particle``.  Consequently,
+substitutions should not be used in docstrings when it is important that
+users have quick access to the full path of the `object` (such as in the
+``See Also`` section).
+
+Templating
+~~~~~~~~~~
+
+Sphinx_ uses the Jinja_ templating engine to generate HTML code. Jinja_
+may be used within the documentation when templating is necessary.
+
+Writing documentation
+=====================
 
 Docstrings
-==========
+----------
 
-* All public classes, methods, and functions should have docstrings.
+A docstring is a comment at the beginning of a function or another
+object that provides information on how to use that function.
+Docstrings begin with ``r"""`` (required when including backslashes,
+such as using LaTeX_ code in equations) or ``"""``, and end with
+``"""``.
```

Comment:
```suggestion
A docstring_ is a comment at the beginning of a function or another
object that provides information on how to use that function
(see `PEP 257 <https://www.python.org/dev/peps/pep-0257/>`_).
Docstrings are designated by surrounding the content with
triple quotes ``"""This is my docstring."""``.  If a docstring
contains math that utilizes LaTex_ code, then we recommend
using a raw-string so you don't have to handle escape
characters on backslashes.  For example, the raw-string
``r""":math:`\alpha`"""`` will render the same as the normal string
``""":math:`\\alpha`"""``.
```

I've already noticed contributors using raw strings when it's not needed.  I think highlighting the difference between raw and normal strings here would be useful.